### DISCUSSION TITLE
Chat text commands visibility

### DISCUSSION THREAD

**Discussion by BearGrilledUK** [sinipelto/dayz-scripts#5] [General]
Hi, I love these commands and scripts, I used them a year ago and using them again now. They're great!

I am wondering about if the chat messages are seen to other players.

1. If I type the '_**/info**_' or '_**/gear**_' commands as an example, will other players see the '_**/info**_' nearby in chat?

2. Secondly, will they see the returning message that shows up in yellow after typing the command? For example, after typing '_**/info**_', will players see the message that says: "_**Players on sever: 2 Player {0} Name: x Pos <x, x, x>...etc**_" in chat?

Thanks!

**Comment by jcaspes** [comment]
Your commands (/gear, /info... ) are visible for players near you, and it is not possible to hide (but it's cool to avoid admin cheat)

But messages are only for you: the method SendPlayerMessage(player, "blabla "); is used to send only to player how type the command

**Reply by BearGrilledUK** [reply]
Thanks for updating me on that, thats great to know! 


I know its probably not possible having looked into it, but would there be a way to add like an Admin freecam command? Where you would type /freecam, because it would be great for identifying cheaters etc.


Also, last thing, in the init.c on line 74. If I changed "admins.txt" to "superadmins.txt", and create a new file in the same location called "superadmins.txt", would this still work? 

Because the idea would be to keep standard admins in the server with the default commands using the #command, but then use the new superadmins to have extra powers that are created in the init.c with the /command.

Thanks a lot!

**Reply by jcaspes** [reply]
You can look my fork of @sinipelto scripts here: https://github.com/jcaspes/dayz-admin-server-scripts
I've added a freecam (/free command) but for this [you need to create a mod](https://github.com/jcaspes/dayz-admin-server-scripts/blob/master/README.md#create-a-mod-for-freecam) for your server with the script i've created for client side. freecam need server to send a command to client that activate the native DayZ Devs freecam so need a mod on client (very simple but mandatory). i know that this change this script idea that there is no mods...

**Reply by BearGrilledUK** [reply]
Wow thats some great work! 

Does that mean that everyone will need to download the mod who connects to the server? or is it just a server side mod that the admins that use the freecam need to download the mod and not normal players? 

Thanks for letting me know!

**Reply by jcaspes** [reply]
a mod always must be on the two sides.
 it's needed at player side because of the freecam command send a message to players side, this message call a code that activate DayZ devs freecam (it's very basic, player must be closed is a room because of not locked so mooving when playing with cam)

**Reply by BearGrilledUK** [reply]
Thanks again for clarifying that, you are very helpful.

I recently had an Idea, I noticed services like CFTools provide a live map of everything on the server. However, I didnt want to add this to my server as I do not trust giving out my Steam profile login details just to login to CFTools website, and that the best features of CFTools are paid for. 

Is there a way, or an already made live map addon/script that can be added to the server?

I found this here: https://github.com/KhaimovMR/DayZLiveMap , but it is so old (11 years) that I do not think it would work anymore. Do you know if this would still work or be possible to make again?

**Reply by jcaspes** [reply]
Hum.... i've codded my own live map :-)
you can see where are players (5 minutes position refresh from logs)
![image](https://github.com/user-attachments/assets/946b238c-fea6-43e5-8503-43ee8982162a)

it store all points / actions (in magenta) so you can see it for each users
![image](https://github.com/user-attachments/assets/816b50d1-fc1a-4d1e-b12e-b146b19cb1ec)

and action can be shown
![image](https://github.com/user-attachments/assets/3b28297e-9216-46d7-ab2f-af5b6c457c6d)

i'ts based on a cron sh script that launch a custom python ADM log parser that generate json files used by a php home made site

you can generate full history from all preserved .ADM files from profile folder...

But i need time to clean the code so it is publishable on github, alos find free images of map because of now used forbidden ones from iZurvive :-), also in DayZ usage condition, thay say it's not allowed tu parse logs... so not sure i can publish that :-)

**Reply by BearGrilledUK** [reply]
Thats some insane level of detail compared to what I have found. Something like that is perfect for identifying cheaters and hackers behaviour on a vanilla server. That is some incredible work! 

There is stuff like iZurvive where you can upload the ADM logs, but it doe not update and does not have the level of details that you have been able to achieve there.

My assumption is that it updates every 5 mins due to the setting adminLogPlayerList = 1; in the serverDZ.cfg that also updates every 5 mins? or maybe that is irrelevant.

Its a shame about the maps that you want to use, I can imagine iZurvive and others dont like anyone else to use their maps for others work. But could you not use the official in-game maps somehow? or maybe they too are forbidden to use in that way... 

Great work though, most of it is above my level haha 😅

**Reply by jcaspes** [reply]
Yes 5 minutes is the player logs, but it is sufficient, and already generate lot of data if there is lot of players...
For the map i need to find one with good resolution for each existing map...
Your question motivates me to try improve the code.. so i can publish it

**Reply by BearGrilledUK** [reply]
I hope you are able to improve and take it further if it is possible. As much as I have looked around, there is nothing much available to anyone out there in as much detail. I wish DayZ had some more support and admin capabilities so everyone doesn't have to get mods just to keep a server safe from cheaters.

**Comment by sinipelto** [comment]
That is good work, thank you for your contribution @jcaspes! I like the separation of concerns you have accomplished splitting the source files into classes, I wasnt aware it would be possible with the scripts. Yes its true it will require a mod, whereas the commands in the main repo has been for pure vanilla so far.

**Reply by jcaspes** [reply]
Yes, i started to split without been sure that it will work... and finally found that *.c files need to be in the root of server files, but it's not a big constraint.
The vanilla style of this script was also the reason i've started to use your script, becasue of my first server was an experimental one because of using a linux server. On experimental servers users must add mods manually so a pure server version was great !
